datalad/customremotes/ria_utils.py

```diff
@@ -0,0 +1,158 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Helper for RIA stores
+
+"""
+
+
+class UnknownLayoutVersion(Exception):
+    pass
+
+
+known_versions_objt = ['1', '2']
+known_versions_dst = ['1']
+
+
+# TODO: This is wrong and should consider both versions (store+dataset)
+def get_layout_locations(version, base_path, dsid):
+    """Return dataset-related path in a RIA store
+
+    Parameters
+    ----------
+    version : int
+      Layout version of the store.
+    base_path : Path
+      Base path of the store.
+    dsid : str
+      Dataset ID
+
+    Returns
+    -------
+    Path, Path, Path
+      The location of the bare dataset repository in the store,
+      the directory with archive files for the dataset, and the
+      annex object directory are return in that order.
+    """
+    if version == 1:
+        dsgit_dir = base_path / dsid[:3] / dsid[3:]
+        archive_dir = dsgit_dir / 'archives'
+        dsobj_dir = dsgit_dir / 'annex' / 'objects'
+        return dsgit_dir, archive_dir, dsobj_dir
+    else:
+        raise ValueError("Unknown layout version: {}".format(version))
+
+
+def verify_ria_url(url, cfg):
+    """Verify and decode ria url
+
+    Expects a ria-URL pointing to a RIA store, applies rewrites and tries to
+    decode potential host and base path for the store from it. Additionally
+    raises if `url` is considered invalid.
+
+    ria+ssh://somehost:/path/to/store
+    ria+file:///path/to/store
+
+    Parameters
+    ----------
+    url : str
+      URL to verify an decode.
+    cfg : dict-like
+      Configuration settings for rewrite_url()
+
+    Raises
+    ------
+    ValueError
+
+    Returns
+    -------
+    tuple
+      (host, base-path, rewritten url)
+    """
+    from datalad.config import rewrite_url
+    from datalad.support.network import URL
```

Comment:
no need to delay the imports... since so prevalent -- is there a shift in coding style to delay all imports until point of use we should all discuss/adopt???